network/peer/tls_config_test.go

```diff
@@ -0,0 +1,131 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package peer_test
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/crypto/ed25519"
+
+	"github.com/ava-labs/avalanchego/network/peer"
+	"github.com/ava-labs/avalanchego/staking"
+)
+
+func TestValidateCertificate(t *testing.T) {
+	for _, testCase := range []struct {
+		description string
+		input       func(t *testing.T) tls.ConnectionState
+		expectedErr error
+	}{
+		{
+			description: "Valid TLS cert",
+			input: func(t *testing.T) tls.ConnectionState {
+				key, err := rsa.GenerateKey(rand.Reader, 2048)
+				require.NoError(t, err)
+				x509Cert := makeRSACertAndKey(t, key)
+				return tls.ConnectionState{PeerCertificates: []*x509.Certificate{&x509Cert.cert}}
+			},
+		},
+		{
+			description: "No TLS certs given",
+			input: func(*testing.T) tls.ConnectionState {
+				return tls.ConnectionState{}
+			},
+			expectedErr: peer.ErrNoCertsSent,
+		},
+		{
+			description: "Empty certificate given by peer",
+			input: func(*testing.T) tls.ConnectionState {
+				return tls.ConnectionState{PeerCertificates: []*x509.Certificate{nil}}
+			},
+			expectedErr: peer.ErrEmptyCert,
+		},
+		{
+			description: "nil RSA key",
+			input: func(t *testing.T) tls.ConnectionState {
+				key, err := rsa.GenerateKey(rand.Reader, 2048)
+				require.NoError(t, err)
+
+				x509CertWithNilPK := makeRSACertAndKey(t, key)
+				x509CertWithNilPK.cert.PublicKey = (*rsa.PublicKey)(nil)
+				return tls.ConnectionState{PeerCertificates: []*x509.Certificate{&x509CertWithNilPK.cert}}
+			},
+			expectedErr: staking.ErrInvalidRSAPublicKey,
+		},
+		{
+			description: "No public key in the cert given",
+			input: func(t *testing.T) tls.ConnectionState {
+				key, err := rsa.GenerateKey(rand.Reader, 2048)
+				require.NoError(t, err)
+
+				x509CertWithNoPK := makeRSACertAndKey(t, key)
+				x509CertWithNoPK.cert.PublicKey = nil
+				return tls.ConnectionState{PeerCertificates: []*x509.Certificate{&x509CertWithNoPK.cert}}
+			},
+			expectedErr: peer.ErrEmptyPublicKey,
+		},
+		{
+			description: "EC cert",
+			input: func(t *testing.T) tls.ConnectionState {
+				ecKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+				require.NoError(t, err)
+
+				basicCert := basicCert()
+				certBytes, err := x509.CreateCertificate(rand.Reader, basicCert, basicCert, &ecKey.PublicKey, ecKey)
+				require.NoError(t, err)
+
+				ecCert, err := x509.ParseCertificate(certBytes)
+				require.NoError(t, err)
+				return tls.ConnectionState{PeerCertificates: []*x509.Certificate{ecCert}}
```

Comment:
Should we add a `makeECDSACert` function as well to align with the `makeRSACert` (`AndKey`) function?

Reply:
No, I'll just make `makeRSACert` handle both.

Reply:
[Done](https://github.com/ava-labs/avalanchego/pull/3501/commits/e45a72ae90a2c993915320789a299e78557fccac#diff-3eb82f83d630d7aeeeba49e200bdb1fad994fff6297717e10dc1faaa54bc0ce6L81-L85)